src/Resources/config/menu.xml

```diff
@@ -44,7 +46,7 @@
             <argument type="collection" />
         </service>
 
-        <service id="knp_menu.menu_provider.lazy" class="Knp\Menu\Provider\LazyProvider" public="false">
+        <service id="knp_menu.menu_provider.lazy" class="%knp_menu.menu_provider.lazy.class%" public="false">
```

Comment:
according to the updated symfony best practices, this should no longer be done. if you want to change how a service is defined, define one with the same name in your application, or write a compiler pass to change all necessary things.

afaik the existing parameters are only here for BC and will be removed in the next major version.